openfoam: restrict the CGAL version compatible with C++14

stephenmsachs opened the pull request:
CGAL throws an [error](https://github.com/CGAL/cgal/blob/50219fc33bc53b5846e9a4257befc6baef3f395a/Installation/include/CGAL/config.h#L147) if C++ lower than 17 is used, while OpenFOAM [forces
C++14](https://develop.openfoam.com/Development/openfoam/-/blob/develop/wmake/rules/General/Gcc/c++?ref_type=heads#L9). This hard C++17 dependency was [introduced](https://github.com/CGAL/cgal/commit/e54408370b2583153dc923884a5ec1587bc7898e) to CGAL version 6.

-------

<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

Commit:
openfoam: restrict the CGAL version compatible with C++14

CGAL throws an
[error](https://github.com/CGAL/cgal/blob/50219fc33bc53b5846e9a4257befc6baef3f395a/Installation/include/CGAL/config.h#L147)
if C++ lower than 17 is used, while OpenFOAM [forces
C++14](https://develop.openfoam.com/Development/openfoam/-/blob/develop/wmake/rules/General/Gcc/c++?ref_type=heads#L9).
This hard C++17 dependency was
[introduced](https://github.com/CGAL/cgal/commit/e54408370b2583153dc923884a5ec1587bc7898e)
to CGAL version 6.

alalazo commented:
@olesenm Can you comment on this one?

wdconinc reviewed (commented):
Sorry for the delay in reviewing. This looks reasonable to me. Can you rebase this PR for newer checks, and evaluate whether anything needs changing since now openfoam [also uses C++17](https://develop.openfoam.com/Development/openfoam/-/commit/44f7a7268a5aec9e6e2ab519dd8f554c97c540d6). Maybe an upper limit can be added to ` depends_on("cgal@:5", when="@2312:")`.

Commit:
Add upper bound since openfoam now uses c++17

https://develop.openfoam.com/Development/openfoam/-/commit/44f7a7268a5aec9e6e2ab519dd8f554c97c540d6

wdconinc reviewed (approved):
Looks good. I'll give @olesenm a few days to chime in.

alecbcs commented:
Going to go ahead and merge after 5 days with no response. @olesenm let me know if you'd like to be removed as a maintainer.
